packages/webapp/components/layouts/PlusLayout/PlusLayout.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { ReactElement, ReactNode, useEffect } from 'react';
+import { cloudinaryPlusBackground } from '@dailydotdev/shared/src/lib/image';
+import { PaymentContextProvider } from '@dailydotdev/shared/src/contexts/PaymentContext';
+import { useAuthContext } from '@dailydotdev/shared/src/contexts/AuthContext';
+import { onboardingUrl } from '@dailydotdev/shared/src/lib/constants';
+import { useRouter } from 'next/router';
+import { useGrowthBookContext } from '@dailydotdev/shared/src/components/GrowthBookProvider';
+import { MainFeedPageProps } from '../MainFeedPage';
+import { PlusHeader } from './PlusHeader';
+
+export default function PlusLayout({
+  children,
+}: MainFeedPageProps): ReactElement {
+  const { user, isAuthReady } = useAuthContext();
+  const { growthbook } = useGrowthBookContext();
+  const router = useRouter();
+
+  const isPageReady = growthbook?.ready && router?.isReady && isAuthReady;
+
+  const shouldRedirectOnboarding = !user && isPageReady;
```

Comment:
Should this not include `isAuthReady`, because router can be ready before boot is fetched

Reply:
naming fooled me